nexus/src/app/sagas/instance_create.rs

```diff
@@ -1009,6 +1014,26 @@ async fn sic_instance_ensure(
     Ok(())
 }
 
+/// Ensure that the necessary v2p mappings exist for this instance
+async fn sic_v2p_ensure(
+    sagactx: NexusActionContext,
+) -> Result<(), ActionError> {
+    let osagactx = sagactx.user_data();
+    let params = sagactx.saga_params::<Params>()?;
+    let opctx = OpContext::for_saga_action(&sagactx, &params.serialized_authn);
+    let instance_id = sagactx.lookup::<Uuid>("instance_id")?;
+
+    // TODO-idempotent if this action fails half way through, unwind is not
+    // called!
```

Comment:
You make a note about this elsewhere, but I'm wondering about unwinding this saga. The way we've structure this, we fail the entire saga node if the request to update mappings on any one sled fails, but without unwinding the mappings we've already sent.

One option would be to put a no-op forward action before this one, that deletes all mappings for an instance. That way we always unwind them, regardless of whether we succeed partway through the following node. I get that there's no actual way to delete these mappings until we fix OPTE#332, but it would structure the saga in a way to alleviate the fact that the `sic_v2p_ensure` action really takes many individual actions.

Reply:
done in 48754ea295053ca3d2d406d6dcc7bdfbc466e192